controller/controller.go

```diff
@@ -154,18 +159,39 @@ func (c *Controller) RunOnce(ctx context.Context) error {
 	return nil
 }
 
-// Run runs RunOnce in a loop with a delay until stopChan receives a value.
-func (c *Controller) Run(ctx context.Context, stopChan <-chan struct{}) {
-	ticker := time.NewTicker(c.Interval)
+// MIN_INTERVAL is used as window for batching events
+const MIN_INTERVAL = 5 * time.Second
```

Comment:
@sheerun I'm not fully in the context but naming is a bit confusing. So `MIN_INTERVAL` is different from `c.Interval` which is controlled by `--interval` CLI, correct ? What happens in `--interval` is less than 5s ?

Reply:
it will batch event less than 4 seconds apart. 5 seconds means that if multiple resources change within 5 seconds, we'll run synchronization just one time

Reply:
--interval from cli is minimum synchronization time, eg. 60s means than we will synchronize at least every 60s, so it's max_interval

Reply:
Got it, thanks :+1: We might want to name these intervals more explicitly and clear in this next PR which will attack MIN_INTERVAL CLI exposure

Reply:
:+1: